uwblueprint/paramedics-react
# Realtime Updates Stage 1

[Notion ticket](https://www.notion.so/uwblueprintexecs/RTU-Stage-1-Refresh-Indicator-75928bea590347b4ac56071d8d0f30c0)
[Figma link](https://www.figma.com/file/IpOPyv79gBwEohvOJvPV6f/%F0%9F%9A%91-Designs-for-Development-(S20%2C-F20)?node-id=1972%3A36028)

Linked to: https://github.com/uwblueprint/paramedics-web/pull/74 (backend setup of graphql subscriptions and websockets)

## Changes
- Front-end updates for the refresh indicator
- Currently increments update count for any changes made to patients in a given ccp
- A few bugs currently exist: changing the URL by rerouting to the patient editing page resets the update counter without updating the patient information to the newest copy. These issues will be resolved with the use of cache updates in stage 2
- NOTE: this PR is a prerequisite to jl-rtu-stage2, which will implement realtime updates directly into the table

## Screenshots
![image](https://user-images.githubusercontent.com/44094521/98450794-c185e080-210d-11eb-958d-407dec46df31.png)

## Testing
Checkout `ct-setup-websockets` branch for `paramedics-web` before testing (https://github.com/uwblueprint/paramedics-web/pull/74)
- Expected behavior: updating information for any existing patient in a CCP should increment the update count for other clients on the same ccp page. (test only on the main ccp dashboard page and patient details dialog)
- go to a ccp populated with patients. Open several clients on that route (ex. http://localhost:3000/events/3/ccps/1)
- update patient details on one client. save changes and check that the badge on the refresh indicator on the top right has incremented. 
- try repeating the above steps with multiple clients 

## Checklist
>[Code review doc for reference](https://www.notion.so/uwblueprintexecs/Code-Review-a21fc85b00394f488e92d9d605f6b2bc)

before opening PR

filling out PR

after opening PR
